src/CakephpRabbitMQ.php

```diff
@@ -0,0 +1,131 @@
+<?php
+namespace RabbitMQ;
+
+use RabbitMQ\Configure\Config;
+use RabbitMQ\Connection\RabbitMQ;
+use RabbitMQ\Helper\ColorfulConsole;
+use RabbitMQ\Helper\MeaningfulTime;
+
+/**
+ * External API
+ */
+class CakephpRabbitMQ
+{
+    /**
+     * Read the configs according to the key provided and listen to them,
+     * listen all if not key is specified
+     *
+     * @param  array $keys
+     * @return void
+     */
+    public static function listen(array $keys = [])
+    {
+        $server = Config::getServer();
+        if (empty($keys)) {
+            $configs = Config::getAllConfigs();
+        } else {
+            $configs = Config::getConfigs($keys);
+        }
+
+        foreach ($configs as $key => $config) {
+            if (empty($keys) || in_array($key, $keys)) {
+                $configs[$key]['_callback'] = static::_callback($key, $config);
+            } else {
+                unset($configs[$key]);
+            }
+        }
+
+        RabbitMQ::listen($server, $configs);
+    }
+
+    /**
+     * Generate internal callback function for queue
+     *
+     * @param  string $key
+     * @param  array  $config
+     * @return function($message)
+     */
+    protected static function _callback(string $key, array $config)
+    {
+        // Generate the callback according to the callback type provided
+        $callback = null;
+        
+        // Callable
+        if (!empty($config['callback'])) {
```

Comment:
this should be separated to method returning callable, so there would be no elseif used. also be more restrictive and throw an exception if $config['callback'] is not callable or none of the callback types is passed